src/os/layer/preset/abstractpresetservice.js

```diff
@@ -0,0 +1,192 @@
+goog.module('os.layer.preset.AbstractPresetService');
+
+
+const IPresetService = goog.requireType('os.layer.preset.IPresetService');
+
+/**
+ * Class to provide stubs and simple implementation of setDefault() and setPublished()
+ *
+ * @abstract
+ * @implements {IPresetService}
+ */
+class AbstractPresetService {
+  /**
+   * @inheritDoc
+   */
+  insert(preset) {
+    return new Promise((resolve) => {
+      resolve(null);
+    });
+  }
+
+  /**
+   * @inheritDoc
+   */
+  update(preset) {
+    return new Promise((resolve) => {
+      resolve(null);
+    });
+  }
+
+  /**
+   * @inheritDoc
+   */
+  remove(preset) {
+    return new Promise((resolve) => {
+      resolve(null);
+    });
+  }
+
+  /**
+   * @inheritDoc
+   */
+  find(search) {
+    return new Promise((resolve) => {
+      resolve(null);
+    });
+  }
+
+  /**
+   * The compiler is wrong; preset, opt_boolean, and the return are exactly as they should be
+   * @suppress {checkTypes}
+   * @inheritDoc
+   */
+  setDefault(preset, opt_boolean) {
+    return new Promise((resolve, reject) => {
+      const isDefault = (opt_boolean !== false);
+
+      if (typeof preset == 'object') {
+        this.setDefaultInternal(preset, isDefault, resolve, reject);
+      } else {
+        this.find(/** @type {osx.layer.PresetSearch} */ ({
+          'id': [preset]
+        })).then(
+            (results) => {
+              if (results && results.length > 0) {
+                this.setDefaultInternal(results[0], isDefault, resolve, reject);
+              } else {
+                reject(`Could not find the preset using id: ${preset}. Did not set to default`);
+              }
+            },
+            (err) => {
+              reject(err);
+            });
+      }
+    });
+  }
+
+  /**
+   * Handle setDefault
+   *
+   * @param {!osx.layer.Preset} preset
+   * @param {!boolean} isDefault
+   * @param {!Function<?>} resolve
+   * @param {!Function<?>} reject
+   * @protected
+   */
+  setDefaultInternal(preset, isDefault, resolve, reject) {
+    const id = preset['id'];
+    const layerFilterKey = preset['layerFilterKey'];
+
+    // loop through all presets in the layer; set the one with ID to isDefault
+    this.find(/** @type {osx.layer.PresetSearch} */ ({
+      'layerFilterKey': [layerFilterKey]
+    })).then(
+        (results) => {
```

Comment:
The deep nesting of complex closures here concerns me a bit. These functions can be defined with typing for their variables like member functions on a class. I think that would help make it clearer what's happening here.

Also there's a bunch of bracket notation references to objects that might be externed types, where . notation will work.